### ISSUE TITLE
Poor webpack own typings detalization

### ISSUE THREAD

**Issue by artem-malko** [webpack/webpack#10828]
# Bug report

**What is the current behavior?**

Loader from `webpack/declarations/WebpackOptions`
```typescript
export interface Loader {
	[k: string]: any;
}
```

Loader from `@types/webpack`
```typescript
interface Loader extends Function {
            (this: LoaderContext, source: string | Buffer, sourceMap?: RawSourceMap): string | Buffer | void | undefined;

            /**
             * The order of chained loaders are always called from right to left.
             * But, in some cases, loaders do not care about the results of the previous loader or the resource.
             * They only care for metadata. The pitch method on the loaders is called from left to right before the loaders are called (from right to left).
             * If a loader delivers a result in the pitch method the process turns around and skips the remaining loaders,
             * continuing with the calls to the more left loaders. data can be passed between pitch and normal call.
             */
            pitch?(remainingRequest: string, precedingRequest: string, data: any): any;

            /**
             * By default, the resource file is treated as utf-8 string and passed as String to the loader.
             * By setting raw to true the loader is passed the raw Buffer.
             * Every loader is allowed to deliver its result as String or as Buffer.
             * The compiler converts them between loaders.
             */
            raw?: boolean;
        }
```

Quite big difference)

As I understand, own types are not so detailed, as in @types/webpack But It will be nice to have the same typings with the same detalization.

**Other relevant information:**
webpack version: beta 15
Node.js version: 10.xxx
Operating System: windows

**Comment by artem-malko** [comment]
There is a temp solutions, who want use webpack beta 15 and rich typings:
tsconfig.json
```json
"compilerOptions": {
    "paths": {
      "webpack": ["./node_modules/@types/webpack/index.d.ts"],
    },
}
```

**Comment by evilebottnawi** [comment]
@artem-malko Something else?

**Comment by artem-malko** [comment]
Emm, why have you marked my last message as spam? It is not possible to use webpack in typescript env in my scenario. This is a simple temp decision(

**Comment by evilebottnawi** [comment]
@artem-malko My mistake, I think it is other thread :smile:

**Comment by artem-malko** [comment]
@evilebottnawi ok)

**Comment by artem-malko** [comment]
> Something else?

What is all in my scenario)

**Comment by melink14** [comment]
Thanks Artem for the work around to use the rich types!
I think the fact that we need to change our typings would be good to add to the migration guide.

For example, for exporting the config producing function which takes env and argv, I found CliConfigOptions was missing and I replaced it with `WebpackOptionsNormalized` but I have no idea if that is correct since I just guessed based on the fact that it contained `mode`. I found no equivalent for `env`.
